engines/e_cng.c

```diff
@@ -0,0 +1,912 @@
+/*
+ * Copyright 2017 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#ifdef _WIN32
+# include <windows.h>
+# include <wincrypt.h>
```

Comment:
MSDN says include ncrypt.h and I don't see no ncrypt.h. Well, nothing prevents wincrypt.h from including ncrypt.h, but then question would be if all SDKs do that... Maybe they all do, did you verify?